src/vs/base/common/assert.ts

```diff
@@ -29,9 +29,25 @@ export function assertNever(value: never, message = 'Unreachable'): never {
 	throw new Error(message);
 }
 
-export function assert(condition: boolean, message = 'unexpected state'): asserts condition {
+/**
+ * Asserts that a condition is `truthy`.
+ *
+ * @throws provided {@linkcode messageOrError} if the {@linkcode condition} is `falsy`.
+ *
+ * @param condition The condition to assert.
+ * @param messageOrError An error message or error object to throw if condition is `falsy`.
+ */
+export function assert(
+	condition: boolean,
+	messageOrError: string | NonNullable<Error> = 'unexpected state',
```

Comment:
What's the benefit of using `NonNullable` on Error?

Reply:
Old habit 🙈 There is no benefit to use it here, removed 👍